src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -56,6 +56,9 @@ internal sealed class Http3RequestStream : IHttpStreamHeadersHandler, IAsyncDisp
         // For the precomputed length case, we need to add the DATA framing for the first write only.
         private bool _singleDataFrameWritten;
 
+        private bool _requestSendCompleted;
+        private bool _responseRecvCompleted;
```

Comment:
How does this differ from `_responseDataPayloadRemaining` and `_requestContentLengthRemaining`?
Just making sure we need these fields and cannot use something we already have.

Reply:
`_responseDataPayloadRemaining` is only for the current DATA frame. `_requestContentLengthRemaining` seems to be usable.

Reply:
Unfortunately, I think we need both actually because with plain `_requestContentLengthRemaining` I can't distinguish failed request sends.